rnd/autogpt_server/autogpt_server/server/server.py

```diff
@@ -103,11 +112,21 @@ def run_service(self):
             endpoint=self.get_graph,
             methods=["GET"],
         )
+        router.add_api_route(
+            path="/graphs/{graph_id}/history",
+            endpoint=self.get_graph_history,
+            methods=["GET"],
+        )
         router.add_api_route(
             path="/graphs",
             endpoint=self.create_new_graph,
             methods=["POST"],
         )
+        router.add_api_route(
+            path="/graphs",
+            endpoint=self.update_graph,
+            methods=["PATCH"],
```

Comment:
`PUT` is more appropriate since the method doesn't allow partial updates:
```suggestion
            methods=["PUT"],
```